mlx/backend/metal/kernels/scaled_dot_product_attention.metal

```diff
@@ -0,0 +1,451 @@
+#include <metal_stdlib>
+#include <metal_simdgroup>
+
+#include "mlx/backend/metal/kernels/scaled_dot_product_attention_params.h"
+using namespace metal;
+
+template<typename T, typename T2, typename T4, uint16_t TILE_SIZE_CONST, uint16_t NSIMDGROUPS>
+[[kernel]] void fast_inference_sdpa_compute_partials_template(const device T *Q [[buffer(0)]],
+                              const device T *K [[buffer(1)]],
+                              const device T *V [[buffer(2)]],
+                              const device uint64_t& L [[buffer(3)]],
+                              const device MLXScaledDotProductAttentionParams& params [[buffer(4)]],
```

Comment:
Not actually a big deal, but this could probably be in the constant namespace

Reply:
yes.  params.KV_TILES would change only every (% tile_size) inferences.  followup change, or would you like it for this one?

Reply:
You can do it later!